### ISSUE TITLE
Allow read_csv to set a single dtype for all columns, or all but certain columns

### ISSUE THREAD

**Issue by Wainberg** [pola-rs/polars#13226]
### Description

`read_csv` doesn't let you set a single dtype for all columns, unless you explicitly specify all the columns by name. For instance, `pl.read_csv({'a': [1], 'b': [2.]}, dtypes=pl.Utf8)` gives the error `TypeError: dtypes should be of type list or dict, got 'DataTypeClass'`. This is inconvenient for biological datasets where you have e.g. 20,000 integer columns and want to load them as `Int32` to save memory.

Another common scenario is when you want to set a dtype for all columns _except_ specified ones. For instance, say you have 20,000 integer columns and one string column (the row names), and you want to load all the integer columns as `Int32`. Pandas's read_csv supports a `defaultdict` for this use-case.

**Comment by ritchie46** [comment]
Workaround in the meantime:

```python
pl.read_csv(file, dtypes={name: pl.Int32 for name, _dt in pl.scan_csv(file, infer_schema_len=0).schema.items()})
```

**Comment by Wainberg** [comment]
@ritchie46 would you be open to a pull request for this? Happy to implement, just need to know the preferred API for it.

**Comment by ritchie46** [comment]
Yes.. Though I'd want something in rust. Maybe a schema mapper with a default type.

**Comment by Wainberg** [comment]
@ritchie46 what would the Python interface look like? Maybe `dtype` for the default dtype (must be a type) and `dtypes` to set dtypes for specific columns (must be a dict of types)? So you could do `pl.read_csv(filename, dtype=pl.Int32, dtypes={'row_name': str})`. I don't think that would be too confusing in practice, as long as there's a descriptive error message when passing a dict for `dtype` or a single type for `dtypes`.
